### PR TITLE
246 refactor make a pivot table between resumes and additional trainings

### PR DESCRIPTION
**Pull request by begoIT24**
We have created a pivot table between resumes and additional_trainings instead of the additional_trainings column in resumes, and have adjusted the model, service, and testing in concordance.
All functionalities are working well.

### CODE REVIEW

File: database/factories/ResumeFactory.php [hunk 1]
```diff
@@ -57,7 +57,7 @@ public function definition(): array
             'project_ids' => json_encode($projectIds),
             'about' => $this->faker->paragraph,
             'modality' => $this->faker->randomElements(['Presencial', 'Híbrid', 'Remot'], rand(1, 3)),
-            'additional_trainings_ids' => json_encode($additionalTrainingsIds),
+           // 'additional_trainings_ids' => json_encode($additionalTrainingsIds),
```
Comment (jordimorillo) [inline database/factories/ResumeFactory.php:60]: Delete, never comment

File: tests/Fixtures/Resumes.php [hunk 2]
```diff
@@ -58,22 +57,22 @@ public static function createResumeWithEmptyProjects(
         $tagsIds = ['tag1', 'tag2'],
         $specialization = 'Frontend',
         $modality = 'Modality',
-        $additionalTrainingsIds = ['additional_training1', 'additional_training2']
+       
     ): Resume {
         $specialization = substr($specialization, 0, 255);
```
Comment (jordimorillo) [inline tests/Fixtures/Resumes.php:63]: Ojo no te cargues las relaciones de los fixtures. Habla con Stephane.

File: app/Models/Resume.php [hunk 3]
```diff
@@ -35,6 +35,6 @@ public function bootcamps(): BelongsToMany
     }
     public function additionalTrainings(): BelongsToMany
     {
-        return $this->belongsToMany(AdditionalTraining::class, 'additional_training_resume');
+        return $this->belongsToMany(AdditionalTraining::class, 'additional_training_resume', 'resume_id', 'additional_training_id');
```
Comment (jordimorillo) [inline app/Models/Resume.php:38]: Comenta con Stephane como lo ha hecho para dejar esto mas limpio

**Review by jordimorillo — CHANGES_REQUESTED** [review]
Review comments

File: database/factories/ResumeFactory.php (hunk 1 above)
Comment (begoIT24) [reply database/factories/ResumeFactory.php]: Done

File: tests/Fixtures/Resumes.php (hunk 2 above)
Comment (begoIT24) [reply tests/Fixtures/Resumes.php:68]: Aclarado

File: app/Models/Resume.php (hunk 3 above)
Comment (begoIT24) [reply app/Models/Resume.php]: Simplificado siguiendo convención de nombres de Laravel

File: app/Models/AdditionalTraining.php [hunk 4]
```diff
@@ -18,7 +18,9 @@ class AdditionalTraining extends Model
```
Comment (jordimorillo) [inline app/Models/AdditionalTraining.php:18]: /**
 * @see https://laravel.com/docs/10.x/eloquent-relationships#many-to-many
 */

File: app/Models/AdditionalTraining.php [hunk 5]
```diff
@@ -18,7 +18,9 @@ class AdditionalTraining extends Model
 
     public function resumes(): BelongsToMany
     {
-        return $this->belongsToMany(Resume::class, 'additional_training_resume');
+         // As a guide, we implement Laravel naming conventions for relationships.
```
Comment (jordimorillo) [inline app/Models/AdditionalTraining.php:21]: Remove this comments

**Review by jordimorillo — APPROVED** [review]
Well done!

**Review by nachoa88 — APPROVED** [review]

File: app/Models/AdditionalTraining.php (hunk 5 above)
Comment (begoIT24) [reply app/Models/AdditionalTraining.php]: done

**Pull request merged**